vaadin-spring/src/main/java/com/vaadin/flow/spring/SpringServlet.java

```diff
@@ -154,7 +153,11 @@ private void setProperty(String property, Properties properties) {
     private void setProperty(String envProperty, String initParam,
             Properties properties) {
         Environment env = context.getBean(Environment.class);
-        String value = env.getProperty(envProperty);
+        String value = env
+                .getProperty(SharedUtil.camelCaseToDashSeparated(envProperty));
```

Comment:
Should the dashseparated value be put to lower case? as the string `vaadin.productionMODE` would become `vaadin.production-mODE` but the environment stores all in lower case.

Reply:
Hmm.....
You are right.
I missed that.

But `vaadin.productionMODE` should become `vaadin.production-m-o-d-e`, not `vaadin.production-mode` .
I will correct the code